core/id/registry.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package id
+
+import "github.com/dolthub/go-mysql-server/sql"
+
+// Operation represents an operation that is being performed or validated.
+type Operation uint8
+
+const (
+	Operation_Rename Operation = iota
+	Operation_Delete
+	Operation_Delete_Cascade
+)
+
+// registry is the implementation of the global registry. This holds all functions that operate or validate a change on
+// an ID.
+type registry struct {
+	performers [][]InternalPerformer
+	validators [][]InternalValidator
+}
+
+// globalRegistry is the variable that is referenced for the registry.
+var globalRegistry = &registry{
+	performers: make([][]InternalPerformer, section_count),
+	validators: make([][]InternalValidator, section_count),
+}
+
+// InternalPerformer is a function that performs the given operation on the original ID. Some operations, such as
+// renames, will use the new ID.
+type InternalPerformer func(ctx *sql.Context, operation Operation, originalID Internal, newID Internal) error
```

Comment:
For this kind of thing, prefer real interfaces to naked funcs. Easy enough to have a `type SequenceIdValidator struct {}` that implements the necessary methods.

Also suggest names: `ChangePerformer` or `OperationPerformer` (and `ChangeValidator` or `OperationValidator` )